src/main/java/com/palantir/docker/compose/DockerComposeRule.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+package com.palantir.docker.compose;
+
+import static java.util.stream.Collectors.toList;
+
+import com.palantir.docker.compose.configuration.DockerComposeFiles;
+import com.palantir.docker.compose.configuration.ProjectName;
+import com.palantir.docker.compose.connection.Container;
+import com.palantir.docker.compose.connection.ContainerCache;
+import com.palantir.docker.compose.connection.DockerMachine;
+import com.palantir.docker.compose.connection.waiting.ServiceWait;
+import com.palantir.docker.compose.connection.waiting.SingleServiceHealthCheck;
+import com.palantir.docker.compose.execution.DefaultDockerCompose;
+import com.palantir.docker.compose.execution.DockerCompose;
+import com.palantir.docker.compose.execution.DockerComposeExecutable;
+import com.palantir.docker.compose.execution.RetryingDockerCompose;
+import com.palantir.docker.compose.logging.DoNothingLogCollector;
+import com.palantir.docker.compose.logging.FileLogCollector;
+import com.palantir.docker.compose.logging.LogCollector;
+import java.io.IOException;
+import java.util.List;
+import org.immutables.value.Value;
+import org.joda.time.Duration;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Value.Immutable
+public abstract class DockerComposeRule extends ExternalResource {
+    private static final Logger log = LoggerFactory.getLogger(DockerComposeRule.class);
+
+    public abstract DockerComposeFiles files();
+
+    @Value.Default
+    protected DockerMachine machine() {
+        return DockerMachine.localMachine().build();
+    }
+
+    @Value.Default
+    public ProjectName projectName() {
+        return ProjectName.random();
+    }
+
+    @Value.Default
+    public DockerComposeExecutable executable() {
+        return DockerComposeExecutable.builder()
+            .dockerComposeFiles(files())
+            .dockerConfiguration(machine())
+            .projectName(projectName())
+            .build();
+    }
+
+    @Value.Default
+    protected DockerCompose dockerCompose() {
+        DockerCompose dockerCompose = new DefaultDockerCompose(executable(), machine());
+        return new RetryingDockerCompose(DockerCompositionBuilder.DEFAULT_RETRY_ATTEMPTS, dockerCompose);
+    }
+
+    @Value.Default
```

Comment:
It was a [constructor argument][1] to the old `DockerComposition` class and I wanted to minimise the number of breaking changes in this PR.  If someone was using their own ContainerCache before, they still can.

If you think it's not sensible behaviour, I can definitely remove it as a plugin point?

[1]:https://github.com/palantir/docker-compose-rule/pull/55/files#diff-d1414afe60d59ac1c4f59661e0b22065L47